src/core/int128.d

```diff
@@ -0,0 +1,696 @@
+{
+    /* Muck about with the signs so we can use the unsigned divide
+     */
+    if (cast(I)c1.msl < 0)
+    {
+        if (cast(I)c2.msl < 0)
+        {
+            Cent r = udiv(neg(c1), neg(c2), modulus);
+            modulus = neg(modulus);
+            return r;
+        }
+        Cent r = neg(udiv(neg(c1), c2, modulus));
+        modulus = neg(modulus);
+        return r;
+    }
+    else if (cast(I)c2.msl < 0)
+    {
+        return neg(udiv(c1, neg(c2), modulus));
+    }
+    else
+        return udiv(c1, c2, modulus);
+}
+
+/****************************
+ * If c1 > c2 unsigned
+ * Params:
+ *      c1 = operand 1
+ *      c2 = operand 2
+ * Returns:
+ *      true if c1 > c2
+ */
+pure
+bool ugt(Cent c1, Cent c2)
+{
+    return (c1.msl == c2.msl) ? (c1.lsl > c2.lsl) : (c1.msl > c2.msl);
+}
+
+/****************************
+ * If c1 >= c2 unsigned
+ * Params:
+ *      c1 = operand 1
+ *      c2 = operand 2
+ * Returns:
+ *      true if c1 >= c2
+ */
+pure
+bool uge(Cent c1, Cent c2)
+{
+    return !ugt(c2, c1);
+}
+
+/****************************
+ * If c1 < c2 unsigned
+ * Params:
+ *      c1 = operand 1
+ *      c2 = operand 2
+ * Returns:
+ *      true if c1 < c2
+ */
+pure
+bool ult(Cent c1, Cent c2)
+{
+    return ugt(c2, c1);
+}
+
+/****************************
+ * If c1 <= c2 unsigned
+ * Params:
+ *      c1 = operand 1
+ *      c2 = operand 2
+ * Returns:
+ *      true if c1 <= c2
+ */
+pure
+bool ule(Cent c1, Cent c2)
+{
+    return !ugt(c1, c2);
+}
+
+/****************************
+ * If c1 > c2 signed
+ * Params:
+ *      c1 = operand 1
+ *      c2 = operand 2
+ * Returns:
+ *      true if c1 > c2
+ */
+pure
+bool gt(Cent c1, Cent c2)
+{
+    if (cast(I)c1.msl >= 0)
+    {
+        if (cast(I)c2.msl >= 0)
+            return ugt(c1, c2);
+        return true;
+    }
+    if (cast(I)c2.msl >= 0)
+        return false;
+    return ugt(c1, c2);
+
+}
+
+/****************************
+ * If c1 >= c2 signed
+ * Params:
+ *      c1 = operand 1
+ *      c2 = operand 2
+ * Returns:
+ *      true if c1 >= c2
+ */
+pure
+bool ge(Cent c1, Cent c2)
+{
+    return !gt(c2, c1);
+}
+
+/****************************
+ * If c1 < c2 signed
+ * Params:
+ *      c1 = operand 1
+ *      c2 = operand 2
+ * Returns:
+ *      true if c1 < c2
+ */
+pure
+bool lt(Cent c1, Cent c2)
+{
+    return gt(c2, c1);
+}
+
+/****************************
+ * If c1 <= c2 signed
+ * Params:
+ *      c1 = operand 1
+ *      c2 = operand 2
+ * Returns:
+ *      true if c1 <= c2
+ */
+pure
+bool le(Cent c1, Cent c2)
+{
+    return !gt(c1, c2);
+}
+
+/****************************
+ * If c1 == c2
+ * Params:
+ *      c1 = operand 1
+ *      c2 = operand 2
+ * Returns:
+ *      true if c1 == c2
+ */
+pure
+bool eq(Cent c1, Cent c2)
+{
+    //printf("eq %d, %d\n", (c1.msl == c2.msl), (c1.lsl == c2.lsl));
+    //printf("eq (%d, %d) (%d, %d)\n", c1.msl, c1.lsl, c2.msl, c2.lsl);
+    return (c1.msl == c2.msl) && (c1.lsl == c2.lsl);
```

Comment:
Why not just `c1 == c2` ? Is the compiler clever enough to generate `PCMPEQQ` instruction, if available? Is this just to avoid calling `opEquals` if an override is available?

Reply:
Why not indeed. You're quite right. Fixed.